lifecycle.md

```diff
@@ -0,0 +1,61 @@
+# Prereq container image lifecycle
+
+The container images in this repository are provided for building .NET source and testing the resultant binaries. The images are built with OS base images and must be updated per the OS lifecycle. This document describes the required patterns.
+
+The images are a mix of Microsoft- and community-supported images.
+
+## Build images
+
+The build images are exclusively built with Azure Linux, supporting both `glibc` and `musl` build flavors. They are built for x64 only, per [Linux build methodology](https://github.com/dotnet/runtime/blob/main/docs/project/linux-build-methodology.md). Microsoft designates the set of images it used for its official build per [The Official Runtime Docker Images](https://github.com/dotnet/runtime/blob/main/docs/workflow/using-docker.md#the-official-runtime-docker-images).
+
+Build images are uniquely created for each new .NET version with the latest Azure Linux available at that time. The image tags include both the given .NET and Azure Linux versions. The tag may also contain a target OS and architecture.
+
+The Azure Linux lifecycle is shorter than the .NET LTS lifecycle. It is expected that build images will need to be (initially) created with Azure Linux n and then replaced with Azure Linux n+1 mid-way through the .NET version lifecycle. They may also need to be updated if the target OS is EOL.
+
+Official build images are deleted after a .NET version is no longer supported. It is possible that images will be required for a while after the published EOL date.
+
+Examples:
+
+- [dotnet-buildtools-prereqs-docker #1263](https://github.com/dotnet/dotnet-buildtools-prereqs-docker/pull/1263)
+- [dotnet/runtime #110198](https://github.com/dotnet/runtime/pull/110198)
+- [dotnet-buildtools-prereqs-docker #1272](https://github.com/dotnet/dotnet-buildtools-prereqs-docker/pull/1272)
+
+## Test images
+
+Test images are produced for the supported Linux distros listed in [Supported OS Versions](https://github.com/dotnet/core/blob/main/os-lifecycle-policy.md) files. In some cases, a [compatible derivative](https://github.com/dotnet/core/blob/main/support.md#compatible-derivatives) will be used. Images are typically built for just `x64` or for all of `Arm32`, `Arm64`, and `x64`. They are shared across .NET versions.
```

Comment:
> Images are typically built for just `x64` or for all of `Arm32`, `Arm64`, and `x64`.

Phrasing is a little awkward. I might just leave it at, we provide a subset of dependent architectures depending on distribution.

Reply:
Fair enough. Works?

```suggestion
Test images are produced for the supported Linux distros listed in [Supported OS Versions](https://github.com/dotnet/core/blob/main/os-lifecycle-policy.md) files. In some cases, a [compatible derivative](https://github.com/dotnet/core/blob/main/support.md#compatible-derivatives) will be used. Images are built for a variety of architectures based on need. They are shared across .NET versions.
```